packages/lib/src/vis/heatmap/utils.ts

```diff
@@ -156,33 +130,27 @@ export function getInterpolator(colorMap: ColorMap, reverse: boolean) {
   return reverse ? (t: number) => interpolator(1 - t) : interpolator;
 }
 
-function getTextureFormatFromType(type: TextureDataType): PixelFormat {
-  if (type === FloatType || type === HalfFloatType) {
-    return RedFormat;
-  }
-
-  if (UNSUPPORTED_TEXTURE_TYPES.has(type)) {
-    throw new Error('Texture type not supported');
-  }
-
-  return RedIntegerFormat;
+export function scaleDomain(
+  domain: Domain,
+  scaleType = ScaleType.Linear
+): Domain {
+  const scaleFn = SCALE_FNS[scaleType];
+  return [scaleFn(domain[0]), scaleFn(domain[1])];
 }
 
 export function getDataTexture(
-  values: NdArray<TextureTypedArray>,
-  textureType = TEXTURE_TYPE_BY_DTYPE[values.dtype],
+  values: NdArray<Float32Array | Uint16Array>,
   magFilter = NearestFilter
 ): DataTexture {
   const { data, shape } = values;
   const [rows, cols] = shape;
-  const format = getTextureFormatFromType(textureType);
 
   return new DataTexture(
     data,
     cols,
     rows,
-    format,
-    textureType,
+    RedFormat,
+    values.dtype === 'uint16' ? HalfFloatType : FloatType,
```

Comment:
Always `RedFormat`, combined with either `FloatType` or `HalfFloatType`, as explained.

Reply:
Could you add some comments to sum up what motivated this choice in the end ? I fear that we might have forgotten if we need to revisit this later

Reply:
Great point. I've added a few paragraphs to explain the reasoning.